mailet/base/src/main/java/org/apache/mailet/base/MailetUtil.java

```diff
@@ -96,19 +96,19 @@ public static String normalizeSubject(String subj, String prefix) {
      * <p>Gets a boolean valued init parameter.</p>
      * @param config not null
      * @param name name of the init parameter to be queried
-     * @return true when the init parameter is <code>true</code> (ignoring case);
-     * false when the init parameter is <code>false</code> (ignoring case);
+     * @return true when the init parameter is <code>true</code>;
+     * false when the init parameter is <code>false</code>;
+     * throw an exception when the value is not of Boolean type;
      * otherwise the default value
      */
     public static Optional<Boolean> getInitParameter(MailetConfig config, String name) {
-        String value = config.getInitParameter(name);
-        if ("true".equalsIgnoreCase(value)) {
-            return Optional.of(true);
+        Object value = config.getInitParameter(name, Boolean.class);
```

Comment:
If we ask for a Boolean, then we got a Boolean, right? No need to type it as an object then cast it?

Reply:
Moreover the entire point of using `Optional` is to not deal with null checks.

I bet this code snipset could actually be as simple as: 

```
return Optional.ofNullable(config.getInitParameter(name, Boolean.class));
```